### ISSUE TITLE
Report HTML injection errors detected by Highlight.js via bugSnag

### ISSUE THREAD

**Issue by joshgoebel** [exercism/exercism#5863]
Jeremy just pointed out we seem to be using BugSnag to track/report errors, and I just looked up that it seems to have a custom reporting capability:

https://docs.bugsnag.com/platforms/javascript/#reporting-handled-errors

There have been several issues now with HTML injection from users code snippets not being properly escaped - that were only caught when the HTML "disappeared" during highlighting... Highlight.js knows about this every time it happens and logs a Javascript console warning (in addition to stripping the HTML - of course any real harm is likely already done) - but I assume these console warnings are just going into the ether on a large production site like Exercism.

Since un-escaped user code displayed as HTML seems to potentially be one of our largest (or at least most visible) injection attack vectors perhaps we could make the reporting proactive... I'd like to propose adding a hook/plugin to Highlight.js that would allow security errors to be hooked - such that with perhaps 5-20 LOC we could channel these errors into BugSnag, complete with the code snippet, what page it appears on, controller/action, etc...

Thoughts?

**Comment by iHiD** [comment]
That sounds good to me, yes. 

You can call `Bugnag.notify` to achieve this: https://github.com/exercism/website/blob/4d2009e76efa318df4e8f4ad9a1a8f4a88dbfa96/app/javascript/components/ErrorBoundary.tsx#L35

**Comment by joshgoebel** [comment]
OH, it looks like it logs every thrown JS Error by default - is that true?  So should we:

- add a custom hook `after:potentialHTMLInjection`and then you write glue to wire that against `Bugnag.notify`
- just raise a `HTMLInjectionError` (or some such) that you could enable with `HLJS.options({securityMode:"error"})` or some such...

Thoughts?

**Comment by iHiD** [comment]
@kntsoriano Can you work with @joshgoebel to decide how to handle this pls.

**Comment by kntsoriano** [comment]
I think the second option should work better since we already report errors by default. I'll have a play and see if it works.

**Comment by joshgoebel** [comment]
Yes, I think that's [hopefully] more generic and could perhaps help more people (with less configuration)... let me know if there is anything specific we should do other than just defining a custom error type and throwing it... 10.3 is around the corner and if we figure this out I could probably get this in that release since it seems fairly trivial so far.

**Comment by kntsoriano** [comment]
@joshgoebel I think that should do the trick. Just raise an error and Bugsnag should report it. We just have one instance of highlightjs that works for the whole site. We should just be able to add the options there and it should all work. Thank you so much :)

**Comment by joshgoebel** [comment]
This will ship in 10.3, perhaps this week.  Still trying to get a few things merged.  I can ping here again after I release.
